### PR TITLE
SDIT-981 return activity details and schedule rules

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityService.kt [hunk 1]
```diff
@@ -118,7 +118,25 @@ class ActivityService(
...
+          internalLocationCode = it.internalLocation?.locationCode,
+          internalLocationDescription = it.internalLocation?.description,
+          capacity = it.capacity ?: 0,
+          description = it.description ?: "",
+          minimumIncentiveLevel = it.iepLevel.code,
+          excludeBankHolidays = it.excludeBankHolidays,
+          scheduleRules = scheduleRuleService.mapRules(it.courseScheduleRules),
+          payRates = listOf(),
+          allocations = listOf(),
+        )
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityService.kt:138]: THe main change. Pay rates and allocations coming next.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourseScheduleRule.kt [hunk 2]
```diff
@@ -64,8 +64,8 @@ data class CourseScheduleRule(
   @Column(nullable = false)
   val startTime: LocalDateTime,
 
-  @Column
-  val endTime: LocalDateTime? = null,
+  @Column(nullable = false)
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourseScheduleRule.kt:67]: Never actually null in the database

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/GetActivityResourceIntTest.kt [hunk 3]
```diff
@@ -95,11 +95,11 @@ class GetActivityResourceIntTest : IntegrationTestBase() {
       fun `finds an active migration with an allocation`() {
         nomisDataBuilder.build {
           programService {
-            courseActivity = courseActivity(startDate = today.toString())
+            courseActivity = courseActivity(startDate = "$today")
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/GetActivityResourceIntTest.kt:98]: A load of these - much prefer this format to remove the noise of toString

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ScheduleRuleService.kt [hunk 4]
```diff
@@ -34,6 +35,21 @@ class ScheduleRuleService {
...
+        endTime = it.endTime.toLocalTime(),
+        monday = it.monday,
+        tuesday = it.tuesday,
+        wednesday = it.wednesday,
+        thursday = it.thursday,
+        friday = it.friday,
+        saturday = it.saturday,
+        sunday = it.sunday,
+      )
+    }
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ScheduleRuleService.kt:51]: This could be a static helper method?

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**